include/xsimd/types/xsimd_batch.hpp

```diff
@@ -537,6 +542,34 @@ namespace xsimd
         return load_unaligned(mem);
     }
 
+    /**
+     * Create a new batch gathering elements starting at address \c src and
+     * offset by each element in \c index. The scale is always 1.
+     * @param src Starting address.
```

Comment:
please add a note here on the fact that if `T != U`, then a `static_cast` is performed